worlds/AutoWorld.py

```diff
@@ -120,7 +120,7 @@ class World(metaclass=AutoWorldRegister):
     """A World object encompasses a game's Items, Locations, Rules and additional data or functionality required.
     A Game should have its own subclass of World in which it defines the required data structures."""
 
-    option_definitions: Dict[str, Option[Any]] = {}  # link your Options mapping
+    option_definitions: Dict[str, AssembleOptions] = {}  # link your Options mapping
```

Comment:
Looking at the hierarchy of worlds.soe.Options.TrapChanceHUD, i don't see Options.AssembleOptions in there, but i do see Options.Option. Without double-checking, I think that'd mean `Type[Option[Any]]` would be the correct type?